packages/nimble-components/src/table/index.ts

```diff
@@ -563,6 +663,63 @@ export class Table<
         this.virtualizer.dataChanged();
     }
 
+    private getTableSelectionState(): TableRowSelectionState {
+        if (this.table.getIsAllRowsSelected()) {
+            return TableRowSelectionState.selected;
+        }
+        if (this.table.getIsSomeRowsSelected()) {
+            return TableRowSelectionState.partiallySelected;
+        }
+        return TableRowSelectionState.notSelected;
+    }
+
+    private getRowSelectionState(
+        row: TanStackRow<TData>
+    ): TableRowSelectionState {
+        if (row.getIsGrouped()) {
+            return this.getGroupedRowSelectionState(row);
+        }
+
+        return row.getIsSelected()
+            ? TableRowSelectionState.selected
+            : TableRowSelectionState.notSelected;
+    }
+
+    private getGroupedRowSelectionState(
+        groupedRow: TanStackRow<TData>
+    ): TableRowSelectionState {
```

Comment:
I presume we needed this implementation to avoid having to call two separate TanStack functions (i.e. `getIsSomeSelected()` and `getIsAllSubRowsSelected()`)? If so, is this call for adding another TanStack function like `getIsGroupRowSelected()` that would return `'all' | 'some' | false`?

Reply:
This was necessitated by TanStack not handling non-selectable group rows well. Therefore, calculating the selection state with nested groups didn't work correctly because all sub-group rows were always considered not selected.

Ultimately, it's probably worth filing/fixing an issue within TanStack, but it hinges on the existing issue I created with them being addressed. So, I wasn't convinced it made sense to try to file that issue until we've reached resolution with the TanStack maintainers on the correct way to have grouped rows that show a selection state but don't maintain a selection state.